arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -2536,6 +2536,41 @@ void RocksDBEngine::pruneWalFiles() {
       << "current number of prunable WAL files: " << _prunableWalFiles.size();
 }
 
+Result RocksDBEngine::dropReplicatedStates(TRI_voc_tick_t databaseId) {
+  rocksdb::ReadOptions readOptions;
+  std::unique_ptr<rocksdb::Iterator> iter(_db->NewIterator(
+      readOptions, RocksDBColumnFamilyManager::get(
+                       RocksDBColumnFamilyManager::Family::Definitions)));
+  auto rSlice = rocksDBSlice(RocksDBEntryType::ReplicatedState);
+  for (iter->Seek(rSlice); iter->Valid() && iter->key().starts_with(rSlice);
+       iter->Next()) {
+    if (databaseId != RocksDBKey::databaseId(iter->key())) {
```

Comment:
It looks like that the database id is always stored in bytes 1 to 9 of the RocksDB key (starting counting bytes at 0).
Right now we are looking for all RocksDB keys that start with `rSlice`, which is a one byte type prefix.
Instead, we could change it to look for keys that have a combined prefix of the type byte **AND** the database id, i.e. check the first 9 bytes.
With that change, we could also use `iterate_upper_bound` in the read options, and set it to prefix **AND** (database id + 1).
Because we are iterating over the definitions column family here and there are only few entries in it, I don't expect a large performance impact here. However, we are looking at more keys than necessary.
I suggest at least leaving a TODO code comment.

Reply:
Good point! I changed it as suggested, and also changed it in `loadReplicatedStates` to do the same.